tests/integration/sync/test_sync_code.py

```diff
@@ -285,11 +285,193 @@ def test_sync_code_shared_codeuri(self):
+
+        # CFN Api call here to collect all the stack resources
+        self.stack_resources = self._get_stacks(TestSyncCodeBase.stack_name)
+        # Lambda Api call here, which tests both the python function and the layer
+        lambda_functions = self.stack_resources.get(AWS_LAMBDA_FUNCTION)
+        for lambda_function in lambda_functions:
+            if lambda_function == "ChildStack/ChildChildStack/HelloWorldFunction":
+                lambda_response = json.loads(self._get_lambda_response(lambda_function))
+                self.assertIn("extra_message", lambda_response)
+                self.assertEqual(lambda_response.get("message"), "12")
+
+    def test_sync_nested_function_layer_race_condition(self):
+        shutil.rmtree(TestSyncCodeBase.temp_dir.joinpath("child_stack").joinpath("child_functions"), ignore_errors=True)
+        shutil.copytree(
+            self.test_data_path.joinpath("nested")
+            .joinpath("before")
+            .joinpath("child_stack")
+            .joinpath("child_functions"),
+            TestSyncCodeBase.temp_dir.joinpath("child_stack").joinpath("child_functions"),
+        )
+        shutil.rmtree(TestSyncCodeBase.temp_dir.joinpath("root_layer"), ignore_errors=True)
+        shutil.copytree(
+            self.test_data_path.joinpath("nested").joinpath("before").joinpath("root_layer"),
+            TestSyncCodeBase.temp_dir.joinpath("root_layer"),
+        )
+        # Run code sync
+        sync_command_list = self.get_sync_command_list(
+            template_file=TestSyncCodeBase.template_path,
+            code=True,
+            watch=False,
+            dependency_layer=True,
+            stack_name=TestSyncCodeBase.stack_name,
+            resource_list=["AWS::Serverless::LayerVersion", "AWS::Serverless::Function"],
+            parameter_overrides="Parameter=Clarity",
+            image_repository=self.ecr_repo_name,
+            s3_prefix=self.s3_prefix,
+            kms_key_id=self.kms_key,
+            tags="integ=true clarity=yes foo_bar=baz",
+        )
+        sync_process_execute = run_command_with_input(sync_command_list, "y\n".encode())
+        self.assertEqual(sync_process_execute.process.returncode, 0)
+
+        # CFN Api call here to collect all the stack resources
+        self.stack_resources = self._get_stacks(TestSyncCodeBase.stack_name)
+        # Lambda Api call here, which tests both the python function and the layer
+        lambda_functions = self.stack_resources.get(AWS_LAMBDA_FUNCTION)
+        for lambda_function in lambda_functions:
+            if lambda_function == "ChildStack/ChildChildStack/HelloWorldFunction":
+                lambda_response = json.loads(self._get_lambda_response(lambda_function))
+                self.assertIn("extra_message", lambda_response)
+                self.assertEqual(lambda_response.get("message"), "10")
+
+    @pytest.mark.skip(reason="Currently not properly supported")
```

Comment:
After the nested support for code sync on Api and Statemachine is merged, this line will be removed.